core/src/subgraph/provider.rs

```diff
@@ -58,6 +59,15 @@ impl<L: LinkResolver> SubgraphProviderTrait for SubgraphProvider<L> {
         Box::new(
             SubgraphManifest::resolve(name.clone(), Link { link }, self.resolver.clone())
                 .map_err(SubgraphProviderError::ResolveError)
+                .and_then(
+                    // Validate the subgraph schema before deploying the subgraph
+                    |subgraph| match validate_schema(&subgraph.schema.document) {
+                        Err(e) => future::err::<_, SubgraphProviderError>(
+                            SubgraphProviderError::SchemaValidationError(e),
+                        ),
+                        _ => future::ok::<_, _>(subgraph),
```

Comment:
These turbofishes seems uncessary, also I wonder If we could just use `Err` and `Ok` instead of `err` and `ok`.

Reply:
You were right. Updated!